src/shared/modules/visualization/visualizationDuck.js

```diff
@@ -38,11 +38,13 @@ function updateLabelData (state, labels) {
 }
 
 function updateStyleData (state, styleData) {
-  return Object.assign({}, state, {styleData: styleData})
+  const clonedStyle = Object.assign({}, styleData)
```

Comment:
I think you can do it and simplifying the updateStyleData function at the same time.

```
const updateStyleData = (state, styleData) => ({...state, styleData: {...styleData}})
```